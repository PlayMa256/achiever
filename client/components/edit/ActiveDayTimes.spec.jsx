import React from 'react';
import { shallow } from 'enzyme';

import ActiveDayTimes from './ActiveDayTimes';

describe('ActiveDayTimes', () => {
	describe('render', () => {
		it('should render 4 TimeGroups', () => {
			const wrapper = shallow(<ActiveDayTimes onTimeChange={() => {}} />);
			expect(wrapper).toMatchSnapshot();

			const wrapperFocus = shallow(<ActiveDayTimes onTimeChange={() => {}} />);
			wrapperFocus.setState({ shouldHaveFocus: { fieldMode: 'hours', index: 1 } });
			expect(wrapper).toMatchSnapshot();

			const { onFocus } = wrapperFocus.find('TimeGroup').at(1).props();
			onFocus('hours');
			expect(wrapperFocus.state('focusedField')).toEqual({ index: 1, fieldMode: 'hours' });
		});
	});
	describe('togglePause', () => {
		it('should hide lunch fields on check', () => {
			const wrapper = shallow(<ActiveDayTimes onTimeChange={() => () => {}} />);
			const { onCheck } = wrapper.find('CheckBox').props();
			onCheck(true);
			expect(wrapper).toMatchSnapshot();
		});
	});
	describe('onChangeTime', () => {
		it('should call onTimeChange with the with correct params', () => {
			const onTimeChangeIntern = jest.fn();
			const onTimeChange = jest.fn(() => onTimeChangeIntern);
			const wrapper = shallow(<ActiveDayTimes onTimeChange={onTimeChange} />);

			const { onSet } = wrapper.find('TimeGroup').at(2).props();
			onSet(10, 10);
			expect(onTimeChangeIntern).toHaveBeenCalledWith(10, 10);
			expect(onTimeChange).toBeCalledWith(2);

			// If no param, should be called with 0
			onSet();
			expect(onTimeChangeIntern).toHaveBeenCalledWith(0, 0);
			expect(onTimeChange).toBeCalledWith(2);
		});
	});
	describe('checksForAutotabNeed', () => {
		describe('_getNextField', () => {
			const defaultCompleteEvent = { target: { value: 12 }, key: 2 };
			const onTimeChangeIntern = jest.fn();
			const onTimeChange = jest.fn(() => onTimeChangeIntern);
			const focusOnSubmit = jest.fn();

			const wrapper = shallow(<ActiveDayTimes
				onTimeChange={onTimeChange}
				focusOnSubmit={focusOnSubmit}
			/>);

			it('should go to next timeGroup if focused on minutes but last', () => {
				const { handleKeyPress } = wrapper.find('TimeGroup').at(2).props();
				// If value is minutes,
				wrapper.setState({ focusedField: { index: 2, fieldMode: 'minutes' } });
				handleKeyPress(defaultCompleteEvent);
				expect(wrapper.state('shouldHaveFocus')).toEqual({ index: 3, fieldMode: 'hours' });
			});
			it('should go to next timeField minutes if focused on hours', () => {
				const { handleKeyPress } = wrapper.find('TimeGroup').at(3).props();
				// If value is hours,
				wrapper.setState({ focusedField: { index: 3, fieldMode: 'hours' } });
				handleKeyPress(defaultCompleteEvent);
				expect(wrapper.state('shouldHaveFocus')).toEqual({ index: 3, fieldMode: 'minutes' });
			});
			it('should focus on submit if is the last minute', () => {
				const { handleKeyPress } = wrapper.find('TimeGroup').at(3).props();
				// If its last field
				wrapper.setState({ focusedField: { index: 3, fieldMode: 'minutes' } });
				handleKeyPress(defaultCompleteEvent);
				expect(focusOnSubmit).toHaveBeenCalled();
			});
		});
		it('should not focus on next field calling onChange', () => {
			const onTimeChangeIntern = jest.fn();
			const onTimeChange = jest.fn(() => onTimeChangeIntern);
			const focusOnSubmit = jest.fn();

			const wrapper = shallow(<ActiveDayTimes
				onTimeChange={onTimeChange}
				focusOnSubmit={focusOnSubmit}
			/>);
			// If value doesn't have legnth 2
			wrapper.setState({ focusedField: { index: 1, fieldMode: 'minutes' } });
			const { onSet: onSet1 } = wrapper.find('TimeGroup').at(1).props();
			onSet1();
			expect(wrapper.state('shouldHaveFocus')).toBeNull();
		});
		it('should not focus if key pressed is not a number', () => {
			const defaultCompleteEvent = { target: { value: 12 }, key: 'a' };
			const onTimeChangeIntern = jest.fn();
			const onTimeChange = jest.fn(() => onTimeChangeIntern);
			const focusOnSubmit = jest.fn();

			const wrapper = shallow(<ActiveDayTimes
				onTimeChange={onTimeChange}
				focusOnSubmit={focusOnSubmit}
			/>);
			// If value doesn't have legnth 2
			wrapper.setState({ focusedField: { index: 1, fieldMode: 'minutes' } });
			const { handleKeyPress } = wrapper.find('TimeGroup').at(1).props();
			handleKeyPress(defaultCompleteEvent);
			expect(wrapper.state('shouldHaveFocus')).toBeFalsy();
		});
		it('should not focus if key input field length is less the 2', () => {
			const defaultCompleteEvent = { target: { value: 2 }, key: '1' };
			const onTimeChangeIntern = jest.fn();
			const onTimeChange = jest.fn(() => onTimeChangeIntern);
			const focusOnSubmit = jest.fn();

			const wrapper = shallow(<ActiveDayTimes
				onTimeChange={onTimeChange}
				focusOnSubmit={focusOnSubmit}
			/>);
			// If value doesn't have legnth 2
			wrapper.setState({ focusedField: { index: 1, fieldMode: 'minutes' } });
			const { handleKeyPress } = wrapper.find('TimeGroup').at(1).props();
			handleKeyPress(defaultCompleteEvent);
			expect(wrapper.state('shouldHaveFocus')).toBeFalsy();
		});
	});
	describe('componentWillReceiveProps', () => {
		it('should mark pause enable if all but break field has content', () => {
			const onTimeChange = jest.fn();
			const focusOnSubmit = jest.fn();

			const wrapper = shallow(<ActiveDayTimes
				onTimeChange={onTimeChange}
				focusOnSubmit={focusOnSubmit}
			/>);
			expect(wrapper.state('pauseIsEnabled')).toBeFalsy();
			wrapper.setProps({
				storedTimes: [
					{ hours: 10, minutes: 0 },
					{},
					{},
					{ hours: 17, minutes: 0 }
				]
			});
			expect(wrapper.state('pauseIsEnabled')).toBeTruthy();
		});
		it('should mark pause disable if any break field has content', () => {
			const onTimeChange = jest.fn();
			const focusOnSubmit = jest.fn();

			const wrapper = shallow(<ActiveDayTimes
				onTimeChange={onTimeChange}
				focusOnSubmit={focusOnSubmit}
			/>);
			expect(wrapper.state('pauseIsEnabled')).toBeFalsy();
			wrapper.setProps({
				storedTimes: [
					{ hours: 10, minutes: 0 },
					{ hours: 10, minutes: '' },
					{},
					{ hours: 17, minutes: 0 }
				]
			});
			expect(wrapper.state('pauseIsEnabled')).toBeFalsy();
		});
	});
});
