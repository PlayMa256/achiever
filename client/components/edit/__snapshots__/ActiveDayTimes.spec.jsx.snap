// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveDayTimes render should render 4 TimeGroups 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveDayTimes
    disabled={false}
    focusOnSubmit={[Function]}
    isHoliday={false}
    onTimeChange={[Function]}
    storedTimes={
      Array [
        Object {},
        Object {},
        Object {},
        Object {},
      ]
    }
    tabIndex={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="no-pause"
        >
          <CheckBox
            disabled={false}
            label="Sem pausa para almoço"
            onCheck={[Function]}
            value={true}
          />
        </div>,
        Array [
          <TimeGroup
            disabled={false}
            emphasis={true}
            hidden={false}
            label="Entrada"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={9}
            shouldHaveFocus={false}
            tabIndexes={5}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={false}
            hidden={true}
            label="Ida para almoço"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={12}
            shouldHaveFocus={false}
            tabIndexes={7}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={false}
            hidden={true}
            label="Volta do almoço"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={13}
            shouldHaveFocus={false}
            tabIndexes={9}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={true}
            hidden={false}
            label="Saída"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={17}
            shouldHaveFocus={false}
            tabIndexes={11}
            time={Object {}}
          />,
        ],
      ],
      "className": "active-day-times",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CheckBox
            disabled={false}
            label="Sem pausa para almoço"
            onCheck={[Function]}
            value={true}
          />,
          "className": "no-pause",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "label": "Sem pausa para almoço",
            "onCheck": [Function],
            "value": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": true,
          "hidden": false,
          "label": "Entrada",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 9,
          "shouldHaveFocus": false,
          "tabIndexes": 5,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "12",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": false,
          "hidden": true,
          "label": "Ida para almoço",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 12,
          "shouldHaveFocus": false,
          "tabIndexes": 7,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "13",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": false,
          "hidden": true,
          "label": "Volta do almoço",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 13,
          "shouldHaveFocus": false,
          "tabIndexes": 9,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "17",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": true,
          "hidden": false,
          "label": "Saída",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 17,
          "shouldHaveFocus": false,
          "tabIndexes": 11,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="no-pause"
          >
            <CheckBox
              disabled={false}
              label="Sem pausa para almoço"
              onCheck={[Function]}
              value={true}
            />
          </div>,
          Array [
            <TimeGroup
              disabled={false}
              emphasis={true}
              hidden={false}
              label="Entrada"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={9}
              shouldHaveFocus={false}
              tabIndexes={5}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={false}
              hidden={true}
              label="Ida para almoço"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={12}
              shouldHaveFocus={false}
              tabIndexes={7}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={false}
              hidden={true}
              label="Volta do almoço"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={13}
              shouldHaveFocus={false}
              tabIndexes={9}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={true}
              hidden={false}
              label="Saída"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={17}
              shouldHaveFocus={false}
              tabIndexes={11}
              time={Object {}}
            />,
          ],
        ],
        "className": "active-day-times",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CheckBox
              disabled={false}
              label="Sem pausa para almoço"
              onCheck={[Function]}
              value={true}
            />,
            "className": "no-pause",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "label": "Sem pausa para almoço",
              "onCheck": [Function],
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": true,
            "hidden": false,
            "label": "Entrada",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 9,
            "shouldHaveFocus": false,
            "tabIndexes": 5,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "12",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": false,
            "hidden": true,
            "label": "Ida para almoço",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 12,
            "shouldHaveFocus": false,
            "tabIndexes": 7,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "13",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": false,
            "hidden": true,
            "label": "Volta do almoço",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 13,
            "shouldHaveFocus": false,
            "tabIndexes": 9,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "17",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": true,
            "hidden": false,
            "label": "Saída",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 17,
            "shouldHaveFocus": false,
            "tabIndexes": 11,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ActiveDayTimes render should render 4 TimeGroups 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveDayTimes
    disabled={false}
    focusOnSubmit={[Function]}
    isHoliday={false}
    onTimeChange={[Function]}
    storedTimes={
      Array [
        Object {},
        Object {},
        Object {},
        Object {},
      ]
    }
    tabIndex={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="no-pause"
        >
          <CheckBox
            disabled={false}
            label="Sem pausa para almoço"
            onCheck={[Function]}
            value={true}
          />
        </div>,
        Array [
          <TimeGroup
            disabled={false}
            emphasis={true}
            hidden={false}
            label="Entrada"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={9}
            shouldHaveFocus={false}
            tabIndexes={5}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={false}
            hidden={true}
            label="Ida para almoço"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={12}
            shouldHaveFocus={false}
            tabIndexes={7}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={false}
            hidden={true}
            label="Volta do almoço"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={13}
            shouldHaveFocus={false}
            tabIndexes={9}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={true}
            hidden={false}
            label="Saída"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={17}
            shouldHaveFocus={false}
            tabIndexes={11}
            time={Object {}}
          />,
        ],
      ],
      "className": "active-day-times",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CheckBox
            disabled={false}
            label="Sem pausa para almoço"
            onCheck={[Function]}
            value={true}
          />,
          "className": "no-pause",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "label": "Sem pausa para almoço",
            "onCheck": [Function],
            "value": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": true,
          "hidden": false,
          "label": "Entrada",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 9,
          "shouldHaveFocus": false,
          "tabIndexes": 5,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "12",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": false,
          "hidden": true,
          "label": "Ida para almoço",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 12,
          "shouldHaveFocus": false,
          "tabIndexes": 7,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "13",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": false,
          "hidden": true,
          "label": "Volta do almoço",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 13,
          "shouldHaveFocus": false,
          "tabIndexes": 9,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "17",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": true,
          "hidden": false,
          "label": "Saída",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 17,
          "shouldHaveFocus": false,
          "tabIndexes": 11,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="no-pause"
          >
            <CheckBox
              disabled={false}
              label="Sem pausa para almoço"
              onCheck={[Function]}
              value={true}
            />
          </div>,
          Array [
            <TimeGroup
              disabled={false}
              emphasis={true}
              hidden={false}
              label="Entrada"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={9}
              shouldHaveFocus={false}
              tabIndexes={5}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={false}
              hidden={true}
              label="Ida para almoço"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={12}
              shouldHaveFocus={false}
              tabIndexes={7}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={false}
              hidden={true}
              label="Volta do almoço"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={13}
              shouldHaveFocus={false}
              tabIndexes={9}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={true}
              hidden={false}
              label="Saída"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={17}
              shouldHaveFocus={false}
              tabIndexes={11}
              time={Object {}}
            />,
          ],
        ],
        "className": "active-day-times",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CheckBox
              disabled={false}
              label="Sem pausa para almoço"
              onCheck={[Function]}
              value={true}
            />,
            "className": "no-pause",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "label": "Sem pausa para almoço",
              "onCheck": [Function],
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": true,
            "hidden": false,
            "label": "Entrada",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 9,
            "shouldHaveFocus": false,
            "tabIndexes": 5,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "12",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": false,
            "hidden": true,
            "label": "Ida para almoço",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 12,
            "shouldHaveFocus": false,
            "tabIndexes": 7,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "13",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": false,
            "hidden": true,
            "label": "Volta do almoço",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 13,
            "shouldHaveFocus": false,
            "tabIndexes": 9,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "17",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": true,
            "hidden": false,
            "label": "Saída",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 17,
            "shouldHaveFocus": false,
            "tabIndexes": 11,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ActiveDayTimes togglePause should hide lunch fields on check 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveDayTimes
    disabled={false}
    focusOnSubmit={[Function]}
    isHoliday={false}
    onTimeChange={[Function]}
    storedTimes={
      Array [
        Object {},
        Object {},
        Object {},
        Object {},
      ]
    }
    tabIndex={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="no-pause"
        >
          <CheckBox
            disabled={false}
            label="Sem pausa para almoço"
            onCheck={[Function]}
            value={true}
          />
        </div>,
        Array [
          <TimeGroup
            disabled={false}
            emphasis={true}
            hidden={false}
            label="Entrada"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={9}
            shouldHaveFocus={false}
            tabIndexes={5}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={false}
            hidden={true}
            label="Ida para almoço"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={12}
            shouldHaveFocus={false}
            tabIndexes={7}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={false}
            hidden={true}
            label="Volta do almoço"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={13}
            shouldHaveFocus={false}
            tabIndexes={9}
            time={Object {}}
          />,
          <TimeGroup
            disabled={false}
            emphasis={true}
            hidden={false}
            label="Saída"
            onFocus={[Function]}
            onSet={[Function]}
            referenceHour={17}
            shouldHaveFocus={false}
            tabIndexes={11}
            time={Object {}}
          />,
        ],
      ],
      "className": "active-day-times",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CheckBox
            disabled={false}
            label="Sem pausa para almoço"
            onCheck={[Function]}
            value={true}
          />,
          "className": "no-pause",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "label": "Sem pausa para almoço",
            "onCheck": [Function],
            "value": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": true,
          "hidden": false,
          "label": "Entrada",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 9,
          "shouldHaveFocus": false,
          "tabIndexes": 5,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "12",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": false,
          "hidden": true,
          "label": "Ida para almoço",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 12,
          "shouldHaveFocus": false,
          "tabIndexes": 7,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "13",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": false,
          "hidden": true,
          "label": "Volta do almoço",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 13,
          "shouldHaveFocus": false,
          "tabIndexes": 9,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "17",
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "emphasis": true,
          "hidden": false,
          "label": "Saída",
          "onFocus": [Function],
          "onSet": [Function],
          "referenceHour": 17,
          "shouldHaveFocus": false,
          "tabIndexes": 11,
          "time": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="no-pause"
          >
            <CheckBox
              disabled={false}
              label="Sem pausa para almoço"
              onCheck={[Function]}
              value={true}
            />
          </div>,
          Array [
            <TimeGroup
              disabled={false}
              emphasis={true}
              hidden={false}
              label="Entrada"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={9}
              shouldHaveFocus={false}
              tabIndexes={5}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={false}
              hidden={true}
              label="Ida para almoço"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={12}
              shouldHaveFocus={false}
              tabIndexes={7}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={false}
              hidden={true}
              label="Volta do almoço"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={13}
              shouldHaveFocus={false}
              tabIndexes={9}
              time={Object {}}
            />,
            <TimeGroup
              disabled={false}
              emphasis={true}
              hidden={false}
              label="Saída"
              onFocus={[Function]}
              onSet={[Function]}
              referenceHour={17}
              shouldHaveFocus={false}
              tabIndexes={11}
              time={Object {}}
            />,
          ],
        ],
        "className": "active-day-times",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CheckBox
              disabled={false}
              label="Sem pausa para almoço"
              onCheck={[Function]}
              value={true}
            />,
            "className": "no-pause",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "label": "Sem pausa para almoço",
              "onCheck": [Function],
              "value": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": true,
            "hidden": false,
            "label": "Entrada",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 9,
            "shouldHaveFocus": false,
            "tabIndexes": 5,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "12",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": false,
            "hidden": true,
            "label": "Ida para almoço",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 12,
            "shouldHaveFocus": false,
            "tabIndexes": 7,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "13",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": false,
            "hidden": true,
            "label": "Volta do almoço",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 13,
            "shouldHaveFocus": false,
            "tabIndexes": 9,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "17",
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "emphasis": true,
            "hidden": false,
            "label": "Saída",
            "onFocus": [Function],
            "onSet": [Function],
            "referenceHour": 17,
            "shouldHaveFocus": false,
            "tabIndexes": 11,
            "time": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
